pinery-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java

```diff
@@ -0,0 +1,1031 @@
+      }
+      String misoType = rs.getString("miso_type");
+      if ("Sample".equals(misoType)) {
+        s.setSampleType(SampleTypeConverter.getSampleType(rs.getString("sampleType")));
+      } else {
+        s.setSampleType(SampleTypeConverter.getNonSampleSampleType(misoType, rs.getString("sampleType_platform"),
+            rs.getString("sampleType_description")));
+      }
+      s.setTissueType(rs.getString("tissueType"));
+      s.setProject(rs.getString("project"));
+      s.setArchived(rs.getBoolean("archived"));
+      s.setCreated(rs.getTimestamp("created"));
+      s.setCreatedById(rs.getInt("createdById"));
+      s.setModified(rs.getTimestamp("modified"));
+      s.setModifiedById(rs.getInt("modifiedById"));
+      s.setTubeBarcode(rs.getString("tubeBarcode"));
+      s.setVolume(rs.getFloat("volume"));
+      if (rs.wasNull()) s.setVolume(null);
+      s.setConcentration(rs.getFloat("concentration"));
+      if (rs.wasNull()) s.setConcentration(null);
+      s.setStorageLocation(extractStorageLocation(rs));
+      PreparationKit kit = new DefaultPreparationKit();
+      kit.setName(rs.getString("kitName"));
+      kit.setDescription(rs.getString("kitDescription"));
+      if (kit.getName() != null || kit.getDescription() != null) {
+        s.setPreparationKit(kit);
+      }
+      Set<Attribute> atts = new HashSet<>();
+      for (AttributeKey possibleAtt : AttributeKey.values()) {
+        Attribute att = possibleAtt.extractAttributeFrom(rs);
+        if (att != null) {
+          atts.add(att);
+        }
+      }
+      if (atts.size() > 0) {
+        s.setAttributes(atts);
+      }
+      Boolean qcPassed = rs.getBoolean("qcPassed");
+      String detailedQcStatus = rs.getString("detailedQcStatus");
+      Status status = new DefaultStatus();
+      status.setState((qcPassed == null || !qcPassed) ? SAMPLE_STATUS_NOT_READY : SAMPLE_STATUS_READY);
+      status.setName(detailedQcStatus == null ? status.getState() : detailedQcStatus);
+      s.setStatus(status);
+      s.setPreMigrationId(rs.getLong("premigration_id"));
+      if (rs.wasNull()) s.setPreMigrationId(null);
+
+      return s;
+    }
+
+    private String extractStorageLocation(ResultSet rs) throws SQLException {
+      String boxAlias = rs.getString("boxAlias");
+      if (boxAlias == null) {
+        if (rs.getBoolean("discarded")) {
+          return "EMPTY";
+        }
+        return null;
+      }
+
+      String boxLocation = rs.getString("boxLocation");
+      String boxPosition = rs.getString("boxPosition");
+
+      StringBuilder sb = new StringBuilder();
+      if (boxLocation != null && !boxLocation.isEmpty()) {
+        sb.append(boxLocation).append(", ");
+      }
+      sb.append(boxAlias).append(", ").append(boxPosition);
+
+      return sb.toString();
+    }
+
+    public static char toRowChar(int row) throws IllegalArgumentException {
+      if (row < 0 || row > 25) throw new RuntimeException("Box row number must be between 0 and 25");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)